### DISCUSSION TITLE
How would I do "npm run build"?

### DISCUSSION THREAD

**Discussion by nick5454** [cake-build/cake#3286] [Extension Q&A] [answered]
I'm using Cake.CoreCLR 0.38.4 and Cake.NPM

I'd like to do npm run build {path}, but my code is not working. Any suggestions

```cakescript
var settings = new NpmInstallSettings 
{
    Production = false,
    WorkingDirectory = "***path***"
};

NpmInstall(settings);

var runSettings = new NpmRunScriptSettings 
{
    ScriptName = "run build",
    LogLevel = NpmLogLevel.Verbose
};

NpmRunScript(runSettings);
```

**Comment by nils-a** [comment, marked as answer]
Have you tried 

```cakescript
   var runSettings = new NpmRunScriptSettings 
   {
      ScriptName = "build",
      LogLevel = NpmLogLevel.Verbose,
   };
   runSettings.Arguments.Add("{path}");
   NpmRunScript(runSettings);
```

**Reply by nick5454** [reply]
That didn't work, is there a command for NPM that would build "npm run build" output

**Reply by nils-a** [reply]
What was the error when that did not work?
Additionally, can you post the output of the Cake run with `--verbosity=diagnostic` this will enable us to see the `npm` statement that is used.

**Reply by augustoproiete** [reply]
> _That didn't work_

What does that mean? Do you get an error (and which error?)

The most basic way to execute `npm run build` is this:

```cakescript
#addin nuget:?package=Cake.Npm&version=1.0.0
NpmRunScript("build");
```

Docs and examples: https://cakebuild.net/extensions/cake-npm/

You can [tell Cake to display the command-line being executed](https://cakebuild.net/docs/running-builds/configuration/default-configuration-values#show-process-command-line).

```powershell
dotnet cake --settings_showprocesscommandline=true
```

Which should display something like:

```
Executing: "C:/Program Files/nodejs/npm.cmd" run-script "build"
```

NB: `run-script` and `run` are synonyms...

**Comment by nick5454** [comment]
Sorry, you are correct. I'm new to nodeJs and npm so I was confused
